### ISSUE TITLE
Track method is polluting/overriding token property in the passed properties argument

### ISSUE THREAD

**Issue by UnGaucho** [mixpanel/mixpanel-js#276]
https://github.com/mixpanel/mixpanel-js/blob/444464f829a237dcd2744d13d1954d899d6c7001/src/mixpanel-core.js#L792

It's bad practice to do this and it's causing other issues with our internal tools.

It should probably either make a copy or wrap it into an internal object. Would a pull request be welcome?

**Comment by tdumitrescu** [comment]
Yes, this is a long-standing open issue (also in https://github.com/mixpanel/mixpanel-js/issues/183 and https://github.com/mixpanel/mixpanel-js/issues/107).

A PR fix would be welcome! Basically just need to replace this initialization: https://github.com/mixpanel/mixpanel-js/blob/444464f829a237dcd2744d13d1954d899d6c7001/src/mixpanel-core.js#L791 with this cloning call: https://github.com/mixpanel/mixpanel-js/blob/444464f829a237dcd2744d13d1954d899d6c7001/src/mixpanel-core.js#L808-L814

**Comment by UnGaucho** [comment]
Oops, should've checked the outstanding issues. Will do.

**Comment by tdumitrescu** [comment]
This should be fixed as of https://github.com/mixpanel/mixpanel-js/releases/tag/v2.48.0
